### ISSUE TITLE
Fails on CSV with large fields

### ISSUE THREAD

**Issue by ShinNoNoir** [harelba/q#97]
For CSV files with large fields, the following error message is displayed:

```
Error('field larger than field limit (131072)',)
```

A solution is to increase the field size limit, [as explained here](http://stackoverflow.com/a/15063941/399990):

```
csv.field_size_limit(sys.maxsize)
```

**Comment by bitti** [comment]
What's more common? That someone really want's to read a CSV which such big fields or that one happens to read a CSV which is wrongly quoted and would rather prefer to get an error message?

**Comment by ShinNoNoir** [comment]
In my case, it's because sometimes one of the fields can be very large. (I'm reading CSV files generated via the Amazon Mechanical Turk platform.)

At the very least, it'd be nice if the field size limit could be configured through a command line flag.

**Comment by harelba** [comment]
Hi,

Checked the _csv C source code and it seems that there should no performance/memory impact for changing this. Also did some performance tests and it seems that there is no real impact.

However, there might be cross platform issues with sys.maxint, and @bitti is right that in most cases it's probably a delimiter issue and not a real field value.

I will catch this kind of error and issue a warning so people can check whether it's a wrong delimiter issue or not, and will either add it as a configurable value or set it to a platform-robust appropriate value.

@ShinNoNoir If you need a modified version in the mean time, please tell me and i'll make one and send it to you over email. Also, it would be very interesting for me to see this kind of data, just so see the actual use-case. If you could send me one or two rows of data over email, that would be great.

Harel

**Comment by ShinNoNoir** [comment]
@harelba: I'm not in an immediate need of a modified version, but thanks for the offer. Also, I'll see if I can produce some sample data. (I'll have to go over it and anonymize some fields.)

**Comment by harelba** [comment]
@ShinNoNoir Thanks, if it's too much effort, then it's ok, i wanted to see it mainly out of curiosity.

**Comment by mahiki** [comment]
Adding to this issue, one common case is where the fields are actually not very large, but there are single quote or double quotes contained in the field. *note to self: always clean data before processing!*

    sed s:\"::g data.txt > tmp ; sed s:\'::g tmp > data.txt

**Comment by ckuenne** [comment]
I would like to bump this issue. It would be great if you could add this variable as a user configurable parameter.

Carsten

**Comment by harelba** [comment]
Fixed in v1.6.3 - The size is now dynamic and can be passed as a parameter. Also, in case of an error the offending line number is being shown.
